lib/hooks.dart

```diff
@@ -101,3 +101,37 @@ StateHook<T> useState<T>(T initialValue) => StateHook(initialValue);
 ///
 /// Learn more: <https://reactjs.org/docs/hooks-reference.html#lazy-initial-state>.
 StateHook<T> useStateLazy<T>(T init()) => StateHook.lazy(init);
+
+/// Returns a memoized version of [callback] that only changes if one of the [dependencies] has changed.
+///
+/// Note there are two rules for using Hooks (<https://reactjs.org/docs/hooks-rules.html>):
+///
+/// * Only call Hooks at the top level.
+/// * Only call Hooks from inside a [DartFunctionComponent].
```

Comment:
not a big deal, and i only just realized this after reviewing all these, that maybe this should be formatted like this: 

```
/// > __Note:__ there are two [rules for using Hooks](https://reactjs.org/docs/hooks-rules.html):
/// >
/// > * Only call Hooks at the top level.
/// > * Only call Hooks from inside a [DartFunctionComponent].
```

Reply:
I updated this in a couple places and I'll make sure the rest are updated in the overall hooks branch